### PR TITLE
NEGF: add new print options

### PR DESCRIPTION
**Pull request by dryndyk**
New sections
NEGF%PRINT%PROGRAM_RUN_INFO
NEGF%PRINT%PROGRAM_RUN_DETAILS
for control of printing levels.
Now "silent" is silent, etc.

CPABORT => CPWARN in src/qs_moments.F:1347,
it blocked print level HIGH for some structures.

### CODE REVIEW

**Comment by dev-zero** [comment]
I think we should rather control verbosity through the section parameter of a print section, rather than extra print section keywords, and keep the print keywords for controlling the printing of concrete additional properties.

Example:

https://github.com/cp2k/cp2k/blob/72339f3ab4a5af43150d59429dc2c871dcab1268/src/qs_active_space_methods.F#L967-L986

**Comment by dryndyk** [comment]
Yes, I agree. It looks better. I will rewrite it. Thanks.

**Comment by dryndyk** [comment]
I have a question. As far as I see, PRINT_LEVEL is used only twice in the code + GLOBAL%PRINT_LEVEL. 
PROGRAM_RUN_INFO is used everywhere.

**Comment by dev-zero** [comment]
I would probably keep the `PROGRAM_RUN_INFO` and get the section parameter describing the level from there

**Comment by dev-zero** [comment]
such that one can do:

```
&NEGF
  &PRINT
    &PROGRAM_RUN_INFO
    &END
  &END
&END
```

for the default, and

```
&NEGF
  &PRINT
    &PROGRAM_RUN_INFO HIGH
    &END
  &END
&END
```

for the more verbose one

**Comment by dev-zero** [comment]
... on the other hand we also have similar flags like in this PR in xtb, dftb and tddpt2 (`DETAILED_ENERGY`), but also explicit `VERBOSITY` flags in transport and linear scaling. We should probably document a standard and harmonize this at some point.

**Comment by dryndyk** [comment]
OK. I will make one iteration to improve and later we can see.

**Comment by dryndyk** [comment]
> such that one can do:
> 
> ```
> &NEGF
>   &PRINT
>     &PROGRAM_RUN_INFO
>     &END
>   &END
> &END
> ```
> 
> for the default, and
> 
> ```
> &NEGF
>   &PRINT
>     &PROGRAM_RUN_INFO HIGH
>     &END
>   &END
> &END
> ```
> 
> for the more verbose one

No, the logic of PROGRAM_RUN_INFO is the opposite. If HIGH it will print something only when GLOBAL%PRINT_LEVEL is HIGH. And it is OK to be consistent with other output. But I want to have the possibility to print more only for NEGF. So, we need another keyword. I think PRINT_LEVEL inside  PROGRAM_RUN_INFO to overwrite  GLOBAL%PRINT_LEVEL with GLOBAL%PRINT_LEVEL as default.

**Comment by dev-zero** [comment]
Yeah, you are right. Well, then let's leave it at this?

**Comment by dryndyk** [comment]
Wait, I will try to add  PROGRAM_RUN_INFO%PRINT_LEVEL instead of PROGRAM_RUN_DETAILS.
Will be better.

**Comment by dryndyk** [comment]
So, is it OK now?

**Comment by dev-zero** [comment]
> So, is it OK now?

LGTM except that `debug1` is not used, right?

**Comment by dryndyk** [comment]
What do you mean by debug1?

File: src/negf_methods.F [hunk 1]
```diff
@@ -194,9 +195,51 @@ SUBROUTINE do_negf(force_env)
...
+            print2 = .FALSE.
+         CASE (high_print_level)
+            print1 = .TRUE.
+            print2 = .TRUE.
+         CASE (debug_print_level)
+            print1 = .TRUE.
+            print2 = .TRUE.
+         CASE DEFAULT
+            !do nothing
+         END SELECT
```
Comment (dev-zero) [inline src/negf_methods.F]:
```suggestion
         SELECT CASE (print_level)
         CASE (high_print_level, debug_print_level)
            verbose_output = .TRUE.
         CASE DEFAULT
            verbose_output = .FALSE.
         END SELECT
```

File: src/negf_methods.F [hunk 2]
```diff
@@ -194,9 +195,51 @@ SUBROUTINE do_negf(force_env)
...
+      END IF
+
+      IF (log_unit > 0) THEN
+         WRITE (log_unit, '(/,T2,A,T62)') "COMPUTE THE RELEVANT HAMILTONIAN MATRICES"
+      END IF
+
       CALL negf_sub_env_create(sub_env, negf_control, blacs_env, global_env%blacs_grid_layout, global_env%blacs_repeatable)
       CALL negf_env_create(negf_env, sub_env, negf_control, force_env, negf_mixing_section, log_unit)
 
+      IF (log_unit > 0 .AND. print2) THEN
```
Comment (dev-zero) [inline src/negf_methods.F]:
```suggestion
      IF (log_unit > 0 .AND. verbose_output) THEN
```

File: src/negf_methods.F [hunk 3]
```diff
@@ -37,10 +37,10 @@ MODULE negf_methods
...
                                               cp_logger_get_default_io_unit,&
                                               cp_logger_type
-   USE cp_output_handling,              ONLY: cp_p_file,&
-                                              cp_print_key_finished_output,&
-                                              cp_print_key_should_output,&
-                                              cp_print_key_unit_nr
+   USE cp_output_handling,              ONLY: &
+        cp_p_file, cp_print_key_finished_output, cp_print_key_should_output, cp_print_key_unit_nr, &
+        debug_print_level, high_print_level, low_print_level, medium_print_level, &
+        silent_print_level
```
Comment (dev-zero) [inline src/negf_methods.F]:
```suggestion
   USE cp_output_handling,              ONLY: cp_p_file,&
                                              cp_print_key_finished_output,&
                                              cp_print_key_should_output,&
                                              cp_print_key_unit_nr,&
                                              debug_print_level,&
                                              high_print_level
```

**Review by dev-zero — COMMENTED** [review]
sorry, that's what I meant. You should be able to directly accept the changes if you agree.

**Comment by dryndyk** [comment]
OK. Looks good!

**Pull request merged**